fuzzers/main.py

```diff
@@ -0,0 +1,842 @@
+                        self.corpus_deleted_metric.inc()
+                    except google.api_core.exceptions.NotFound:
+                        pass  # Ignore, as it'd mean the file isn't there already
+
+
+class FuzzProcess:
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(
+        self,
+        *,
+        corpus_vers: str,
+        branch: BranchType,
+        target: TargetType,
+        repo_dir: pathlib.Path,
+        log_relpath: pathlib.Path,
+        log_fullpath: pathlib.Path,
+    ):
+        """
+        Create a FuzzProcess object. It can be built with `.build()` and then started with
+        `.start()`.
+
+        `corpus_vers` is the version of the corpus (first path item), used only for logging.
+        `branch` and `target` are the configuration parameters to run this process with.
+        `repo_dir` is the path to the nearcore checkout root.
+        `log_relpath` is the relative path from the log folder to the log file.
+        `log_fullpath` is the absolute path to the log file.s
+        """
+
+        self.corpus_vers = corpus_vers
+        self.branch = branch
+        self.target = target
+        self.repo_dir = repo_dir
+        self.log_relpath = log_relpath
+        self.log_fullpath = log_fullpath
+        self.log_file = open(log_fullpath, 'a', encoding='utf-8')  # pylint: disable=consider-using-with
+
+        self.last_time = 0.
+        self.proc: typing.Any = None  # There's some weirdness around brackets and Popen
+
+        self.fuzz_build_time_metric = FUZZ_BUILD_TIME.labels(
+            branch['name'], target['crate'], target['runner'])
+        self.fuzz_time_metric = FUZZ_TIME.labels(branch['name'],
+                                                 target['crate'],
+                                                 target['runner'],
+                                                 target['flags'])
+        self.fuzz_crashes_metric = FUZZ_TIME.labels(branch['name'],
+                                                    target['crate'],
+                                                    target['runner'],
+                                                    target['flags'])
+
+    def build(self) -> None:
+        """
+        Build the fuzzer runner.
+
+        This is a synchronous operation for convenience reasons, as cargo itself uses all the
+        available cores most of the time it shouldn't be a big deal. The only drawback is that
+        requests to pause/exit the fuzzer would block until the current build is completed.
+        """
+        print(f'Building fuzzer for branch {self.branch} and target '
+              f'{self.target}, log is at {self.log_relpath}', file=sys.stderr)
+
+        # Log metadata information
+        current_commit = str(
+            subprocess.run(
+                ['git', 'rev-parse', 'HEAD'],
+                cwd=self.repo_dir,
+                capture_output=True,
+                check=True,
+            ).stdout)
```

Comment:
That’s a rather convoluted way to write:

```python
        current_commit = subprocess.check_output(
            ('git', 'rev-parse', 'HEAD'),
            cwd=self.repo_dir,
            encoding='utf-8').stdout.strip()
```

;)

And you also want `strip` to get rid of trailing new line.